add layer example for sticky header (#2470)
grommet/hpe-design-system: aries-site/src/pages/components/layer.mdx

Actionable Layers contain the following elements and interactions: 
- A cancel or close button will always be positioned within the footer of the Layer. Note that the "X" icon is omitted to avoid any confusion the user may have on which option they should use to close out the Layer. 

> - For situations where the contents exceed the height of the layer, the header should either always stick atop the Layer while the content scrolls beneath, or to maximize screen real-estate, the header may scroll away as the user scrolls down the content, but re-appear as the user scrolls up, signaling their intent to return to the top of the content.

Reviewer: Can we link to each of the examples in this?
```suggestion
- The Layer header should be sticky while the rest of the Layer content is scrollable. This can either be achieved with a [persistent sticky header]() or a [header that is sticky when the user scrolls up](). 
```

>
  <LayerCenterExample />
</Example>

### Layer Form with sticky Header

The Layer header and action buttons can be sticky while the rest of the Layer content is scrollable.

<Example
  docs="https://v2.grommet.io/layer?theme=hpe#props"
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/layer/LayerFormExample.js"
  height={{ min: 'small' }}
>
  <LayerFormExample />
</Example>

### Layer Form with hide/show sticky Header

> In order maximize screen real-estate, the header may scroll away as the user scrolls down the content, but re-appear as the user scrolls up.

Reviewer: ```suggestion
In order to maximize screen real-estate when content is scrollable, the header may scroll away as the user scrolls down but reappear as the user scrolls up.
```

<Example
  docs="https://v2.grommet.io/layer?theme=hpe#props"

>   code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/layer/LayerFormExample.js"

Reviewer: ```suggestion
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/layer/LayerStickyFormExample.js"
```

>
  <LayerCenterExample />
</Example>

### Layer Form with sticky Header

The Layer header and action buttons can be sticky while the rest of the Layer content is scrollable. This can be accomblished by wrapping a Form within a Box with `overflow` set to `auto`.

<Example
  docs="https://v2.grommet.io/layer?theme=hpe#props"
  code={[
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/layer/LayerStickyExample.js",
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/layer/MonitorFormExample.js"
  ]}
  height={{ min: 'small' }}
>
  <LayerStickyExample />
</Example>

### Layer Form with hide/show sticky Header

> In order to maximize screen real-estate when content is scrollable, the header may scroll away as the user scrolls down but reappear as the user scrolls up.

Reviewer: Should we be mentioning using Header with `sticky="scrollup"` ?

Author: done
